ui/src/dashboards/components/QueryTextArea.tsx

```diff
@@ -28,27 +48,78 @@ class QueryTextArea extends Component {
     }
   }
 
-  componentWillReceiveProps(nextProps) {
+  public render() {
+    const {
+      config: {status},
+    } = this.props
+    const {
+      value,
+      isTemplating,
+      selectedTemplate,
+      filteredTemplates,
+    } = this.state
+
+    return (
+      <div className="query-editor">
+        <textarea
+          className="query-editor--field"
+          onChange={this.handleChange}
+          onKeyDown={this.handleKeyDown}
+          onBlur={this.handleUpdate}
+          ref={this.handleTextAreaRef}
```

Comment:
Could we change this to use a react 16 refObject?

Reply:
thanks! done.